### PR TITLE
gradle 8.3, supporting java-21 via its kotlin dependency

### PR DESCRIPTION
**Pull request by soloturn**
permits to remove hardcoded kotlin version again, like mentioned in #5130.

### CODE REVIEW

**Comment by soloturn** [comment]
tested with java-21, still good.

**Comment by soloturn** [comment]
@BenjaminAmos am i allowed to ping you here as well?

File: gradle/wrapper/gradle-wrapper.properties
Comment (BenjaminAmos) [inline gradle/wrapper/gradle-wrapper.properties]: Although it is larger, we tend to use the `-all` gradle distribution because it makes the IDE tooling more useful.

Comment (BenjaminAmos) [reply gradle/wrapper/gradle-wrapper.properties]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-all.zip
```

**Review by BenjaminAmos — CHANGES_REQUESTED** [review]
We tend to use the `gradle-all` distribution over the `gradle-bin` distribution because it provides better IDE support.

File: gradle/wrapper/gradle-wrapper.properties
Comment (soloturn) [reply gradle/wrapper/gradle-wrapper.properties]: done

**Review by BenjaminAmos — APPROVED** [review]
These changes seem fine to me.

I am approving this but will probably wait for a second opinion before merging.

**Comment by BenjaminAmos** [comment]
To clarify the title here, this will probably only support Java 20. Running on Java 21 will likely need Gradle 8.4 (which is not yet released). It's still an improvement.

**Comment by jdrueckert** [comment]
should we adjust the title in that case @BenjaminAmos ?

**Comment by soloturn** [comment]
am running java-21.

**Pull request merged**
